modules/datasets/samples/pd_inria.cpp

```diff
@@ -53,14 +64,47 @@ int main(int argc, char *argv[])
             cout << "  - ymin = " << y << endl;
             cout << "  - xmax = " << example->bndboxes[j].width + x << endl;
             cout << "  - ymax = " << example->bndboxes[j].height + y << endl;
+
+            Rect r = example->bndboxes[j];
+
+            Rect r0 = r;
+            int vpadding = cvRound(padding * r.height / rheight);
+            r.y = r.y - vpadding;
+            r.height = 2 * vpadding + r.height;
+            r.x = r.x + (r.width / 2);
+            r.width = r.height * rwidth / rheight;
+            r.x = r.x - (r.width / 2);
+
+            if (r0.width > r.width)
+            {
+                r0.x += ((r0.width - r.width)+1) / 2;
+                r0.width = r.width;
+            }
+
+            if(savebbox)
+            {
+                Mat dst;
+                int top = r0.y - r.y;
+                int bottom = top;
+                int left = r0.x - r.x;
+                int right = left;
+
+                copyMakeBorder(img(r0), dst, top, bottom, left, right, BORDER_REFLECT);
```

Comment:
Why do we not just using `vpadding, vpadding, padding, padding` (without calculations with rectangles above)?
Or with scaled padding: `padding *= rwidth / r.width; vpadding *= rheight / r.height` (if applicable, it is not clear from command line description and why it is not zero by default)